src/main/java/com/kainos/ea/dao/CompetencyPerBandLvlDao.java

```diff
@@ -0,0 +1,44 @@
+package com.kainos.ea.dao;
+
+import com.kainos.ea.exception.CompetencyPerBandLvlNotExistException;
+import com.kainos.ea.model.CompetencyPerBandLvl;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompetencyPerBandLvlDao {
+    public CompetencyPerBandLvlDao(){
+    }
+    public List<CompetencyPerBandLvl> getAllCompetencyPerBandLvl(Connection c, int role_id) throws SQLException, CompetencyPerBandLvlNotExistException {
+
+        try{
+            Statement st = c.createStatement();
+            ResultSet rs = st.executeQuery("SELECT DISTINCT temp.id, band_level, band_name, competency_name FROM (\n" +
```

Comment:
by taking in the band id in the URL this query should be much simpler

Reply:
done